decidim-api/lib/decidim/api/test/type_context.rb

```diff
@@ -31,18 +31,19 @@
   def raise_proper_error(error)
     code = error.dig("extensions", "code")
 
-    case code
-    when "NO_PERMISSION_SET"
-      raise Decidim::Api::Errors::PermissionNotSetError, error["message"]
-    when "NOT_FOUND"
-      raise Decidim::Api::Errors::NotFoundError, error["message"]
-    when "NO_FIELD_PERMISSION"
-      raise Decidim::Api::Errors::UnauthorizedFieldError, error["message"]
-    when "NO_OBJECT_PERMISSION"
-      raise Decidim::Api::Errors::UnauthorizedObjectError, error["message"]
-    else
-      raise StandardError, error["message"]
-    end
+    raise "Decidim::Api::Errors::#{code.downcase.classify}".constantize, error["message"] if %w(
```

Comment:
I had to double check this. Let's add a quick comment for dumb people as me 

```suggestion
    # Matches the error code with the Error class
    # For instance, if the error code is NOT_FOUND_ERROR then it will raise the "Decidim::Api::Errors::NotFoundError" class
    raise "Decidim::Api::Errors::#{code.downcase.classify}".constantize, error["message"] if %w(
```